### ISSUE TITLE
About an interesting things

### ISSUE THREAD

**Issue by Endless-Hao** [cardwing/Codes-for-Lane-Detection#25]
@cardwing  Hello，i find an interesting thing. In my own computer(1080ti 11G), I try to change different tensorflow such as 1.4, 1.7... and I also re-install cudu9, Nvidia driver and cudnn. I all get the not good results. But on another computer in my laboratory, the result is ok.

**Comment by Endless-Hao** [comment]
@cardwing  On the computer which can test to good results, I also find that the results were different. The BN Layer in your code divided into test or train. So where the parameters are not fixed, after train all the parameters including BN parameters are fixed I think.

**Comment by Endless-Hao** [comment]
@cardwing  Hey，I seem to find the problem although  I don’t know why. Today I ran all the test sets on the computer which I got good results yesterday,  because I just ran a sequence yesterday, and then I got bad results. I use all the test sets and get bad results. Then I ran a sequence on a computer that I use before (my own 1080TI ) and found the result to be good. Maybe is a memory problem. Running a sequence of results is good, running the entire test set to get weird results.

**Comment by cardwing** [comment]
@Endless-Hao, what do you mean by running a sequence? In the testing phase, 8 video frames are sent to the model every episode.

**Comment by Endless-Hao** [comment]
@cardwing  In the testing process, if I test all the test data I got the not good result. If I test a sequence from the test data I got the good results. As simple to say, divided the test data and test one of them then test another one.  For example, test.txt  only have the files in sequence "05251517_0433.MP4", the result is good. But if test.txt  has all data, the result is not good.
Like below:
Just one sequence:
![00000 results](https://user-images.githubusercontent.com/19277849/50266444-810c1380-045e-11e9-9693-3e01c4028923.png)

All sequence:
![00000 results](https://user-images.githubusercontent.com/19277849/50266461-97b26a80-045e-11e9-97cc-c79df1a665fc.png)

**Comment by cardwing** [comment]
@Endless-Hao, interesting. Maybe it is caused by the bug of tensorflow itself. You can report more results here once you find.

**Comment by Endless-Hao** [comment]
@cardwing  OK, thank you for your help.

**Comment by Endless-Hao** [comment]
@cardwing  what is you cuda version?

**Comment by cardwing** [comment]
@Endless-Hao, cuda 8.0 and cudnn 5.

**Comment by Endless-Hao** [comment]
@cardwing  your new code has some problems, in tes_lanenet.py 
net=lanenet_merge......
net.inference(...)...
have unexpected keyword argument 'pause' 'name'

**Comment by Endless-Hao** [comment]
@cardwing afternoon I try many times test, also in sequences. I find the good results in the morning is an accident. I try again,  the different results also not good. It drives me crazy. Every time result are different ,some good and next would be not good. Also test in different computer (P100 ) get same condition.

**Comment by cardwing** [comment]
@Endless-Hao, you can go to previous commits and select the old version which will not have errors. As to the problem of different outputs, I have no idea since this problem can not be reproduced in my local server.

**Comment by yujincheng08** [comment]
It's because the test data set processor randomizes the images.
See: https://github.com/cardwing/Codes-for-Lane-Detection/blob/master/SCNN-Tensorflow/lane-detection-model/data_provider/lanenet_data_processor_test.py#L33

**Comment by Endless-Hao** [comment]
@yujincheng08  Yes, you are right. The results would not change again. Thank you. The problem is solved.

**Comment by ktr-hubrt** [comment]
> @yujincheng08 Yes, you are right. The results would not change again. Thank you. The problem is solved.

 I also train a scnn model ,and i only get the f1-score of 67,how about you?

**Comment by yujincheng08** [comment]
> > @yujincheng08 Yes, you are right. The results would not change again. Thank you. The problem is solved.
> 
> I also train a scnn model ,and i only get the f1-score of 67,how about you?

Mine works perfectly now.

**Comment by ktr-hubrt** [comment]
> > > @yujincheng08 Yes, you are right. The results would not change again. Thank you. The problem is solved.
> > 
> > 
> > I also train a scnn model ,and i only get the f1-score of 67,how about you?
> 
> Mine works perfectly now.

So, i understand when you use multi-gou training code, F1-score of your result can reach 71?

**Comment by ktr-hubrt** [comment]
> > > @yujincheng08 Yes, you are right. The results would not change again. Thank you. The problem is solved.
> > 
> > 
> > I also train a scnn model ,and i only get the f1-score of 67,how about you?
> 
> Mine works perfectly now.

could you share your training details with me?

**Comment by sainisanjay** [comment]
@Endless-Hao @cardwing, May i know how to visualise the final results. After running `test_lanenet.py` it generates probability maps (`00000_1_avg.png`) and `00000.exist.txt`. Could you tell me how to visualise these results?

**Comment by yujincheng08** [comment]
@sainisanjay https://github.com/XingangPan/SCNN/tree/master/tools/prob2lines see these code. It can be easily transferred to python.

**Comment by sainisanjay** [comment]
Thanks @yujincheng08

**Comment by snarb** [comment]
https://github.com/cardwing/Codes-for-Lane-Detection/blob/master/ERFNet-CULane-PyTorch/prob_to_lines.py it is already transferred
